src/syntax/ppx_sedlex.ml

```diff
@@ -199,7 +201,10 @@ let partition (name, p) =
         [%expr Char.code (String.get [%e evar (table_name t)] [%e c]) - 1]
   in
   let body = gen_tree (decision_table p) in
-  glb_value name (func [pvar "c", body])
+  glb_value name (func [(pconstr "Some" [pvar "uc"],
+                         [%expr let c = Uchar.to_int uc in [%e body]]);
+                        (pconstr "None" [],
+                         [%expr let c = (-1) in [%e body]])]) (* XXX *)
```

Comment:
The duplication is fine, you can remove the comments.

Reply:
Gotcha.

Reply:
I'm going to leave the explanation but not have it indicate that this is a problem, vis:

`(* We duplicate the body for the EOF (-1) case rather than creating
   an interior utility function. *)`

I've also removed the `(* XXX *)` comment.